test/SaveMethodCrashTest.cpp

```diff
@@ -0,0 +1,196 @@
+    bool isTreeWidgetItemValid(QTreeWidgetItem* item, QTreeWidget* tree);
+    bool validateCurrentItem(QTreeWidgetItem*& currentItem, QTreeWidget* tree);
+};
+
+bool SaveMethodCrashTest::unfixedSaveMethod(QTreeWidgetItem* currentItem)
+{
+    // This is the UNSAFE pattern that caused the crash
+    if (!currentItem) {
+        return false;
+    }
+    
+    // DANGER: This line would cause heap-use-after-free if currentItem
+    // points to freed memory after tree clearing
+    // Note: For top-level items, parent() returns nullptr, so we'll check differently
+    QString itemText = currentItem->text(0);  // ← CRASH HERE in unfixed version if freed
+    return true;
+}
+
+bool SaveMethodCrashTest::fixedSaveMethod(QTreeWidgetItem*& currentItem, QTreeWidget* tree)
+{
+    // This is the SAFE pattern after our fix
+    if (!currentItem) {
+        return false;
+    }
+    
+    // SAFE: Validate the pointer before use
+    if (!validateCurrentItem(currentItem, tree)) {
+        return false;  // Pointer was invalid and has been cleared
+    }
+    
+    // Now safe to use currentItem 
+    QString itemText = currentItem->text(0);
+    return true;
+}
+
+bool SaveMethodCrashTest::isTreeWidgetItemValid(QTreeWidgetItem* item, QTreeWidget* tree)
+{
+    if (!item || !tree) {
+        return false;
+    }
+
+    // Check if item is a top-level item
+    for (int i = 0; i < tree->topLevelItemCount(); ++i) {
+        if (tree->topLevelItem(i) == item) {
+            return true;
+        }
+    }
+
+    // Check if it's a child item recursively
+    std::function<bool(QTreeWidgetItem*)> findItemRecursive = [&](QTreeWidgetItem* parent) -> bool {
+        for (int i = 0; i < parent->childCount(); ++i) {
+            QTreeWidgetItem* child = parent->child(i);
+            if (child == item) {
+                return true;
+            }
+            if (findItemRecursive(child)) {
+                return true;
+            }
+        }
+        return false;
+    };
+
+    for (int i = 0; i < tree->topLevelItemCount(); ++i) {
+        if (findItemRecursive(tree->topLevelItem(i))) {
+            return true;
+        }
+    }
+
+    return false;
+}
+
+bool SaveMethodCrashTest::validateCurrentItem(QTreeWidgetItem*& currentItem, QTreeWidget* tree)
+{
+    if (!currentItem || !tree) {
+        currentItem = nullptr;
+        return false;
+    }
+
+    if (!isTreeWidgetItemValid(currentItem, tree)) {
+        currentItem = nullptr;  // Clear invalid pointer - this is the key fix
+        return false;
+    }
+
+    return true;
+}
+
+void SaveMethodCrashTest::testSaveMethodWithClearedTree()
+{
+    // Create tree and items
+    QTreeWidget* tree = new QTreeWidget();
+    QTreeWidgetItem* item1 = new QTreeWidgetItem(tree, QStringList("Test Item 1"));
+    QTreeWidgetItem* item2 = new QTreeWidgetItem(tree, QStringList("Test Item 2"));
+    
+    // Simulate having a current item pointer (like mpCurrentTriggerItem)
```

Comment:
This test just simulates a crash and simulates fixing a crash, similar to another test we looked at recently. I'm not sure I see the value in this. I would understand it more if the test actually triggered a crash instead of simulating it. Can you help explain? I also don't mind removing the test.